src/openms/source/DATASTRUCTURES/DataValue.cpp

```diff
@@ -676,7 +675,7 @@ namespace OpenMS
 
   bool operator==(const DataValue& a, const  DataValue& b)
   {
-    if (a.value_type_ == b.value_type_)
+    if (a.value_type_ == b.value_type_ && a.unit_type_ == b.unit_type_ && a.unit_ == b.unit_)
```

Comment:
now we actually also compare the unit, which led to a lot of test failures since we *forgot* to carry the unit in `userParam`